### PR TITLE
steve exercise 2

### PR DESCRIPTION
**Pull request by stevemao**

### CODE REVIEW

**Comment by stevemao** [comment]
Unfortunately this solution doesn't seem to be as simple as a for loop :( (assume we won't change the requirement anymore in the future.)
Would be good to see a solution that's as easy as for loop and still readable and maintainable.

File: cis194/week3/stevemao/Golf.hs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+removeFirst :: [a] -> [a]
+removeFirst (_: xs) = xs
+
+getSecondIfMax :: (Integer, Integer, Integer) -> [Integer] -> [Integer]
+getSecondIfMax (a, b, c) list
+  | b > a && b > c = [b] ++ list
+  | otherwise = list
+
+localMaxima :: [Integer] -> [Integer]
+localMaxima list = foldr getSecondIfMax [] (zip3 list (removeFirst list) (removeFirst . removeFirst $ list))
```
Comment (sinogermany) [inline cis194/week3/stevemao/Golf.hs]:
```suggestion
localMaxima list = foldr getSecondIfMax [] $ zip3 list (drop 1 list) (drop 2 list)
```

**Comment by sinogermany** [comment]
[cis194/week3/stevemao/Main](https://github.com/fp-works/2019-winter-Haskell-school/pull/42/files#diff-2ba0860ed37ec58d692bf58585b06112) can be git ignored

File: cis194/week3/stevemao/Golf.hs (hunk 1 above)
Comment (stevemao) [reply cis194/week3/stevemao/Golf.hs]: Thanks @sinogermany

**Pull request merged**